src/AppBundle/Command/ImportSaleCommand.php

```diff
@@ -83,21 +81,13 @@ private function getSales($limit, $offset)
         return $response->json();
     }
 
-    private function saveSales($jsonSales, $existingInvoices)
+    private function saveSales($jsonSales, $repository)
     {
-        foreach ($jsonSales as $jsonSale) {
-            $exists = false;
-            foreach ($existingInvoices as $existing) {
-                if (
-                    $existing['invoice'] == $jsonSale['invoice'] &&
-                    $existing['licenseId'] == $jsonSale['licenseId'] &&
-                    $existing['pluginKey'] == $jsonSale['pluginKey']
-                ) {
-                    $exists = true;
-                }
-            }
+        $newCnt = 0;
 
-            if (!$exists) {
+        foreach ($jsonSales as $jsonSale) {
+            if ($repository->findIfSaleIsNew($jsonSale['invoice'], $jsonSale['licenseId'], $jsonSale['pluginKey'])) {
```

Comment:
I have performance concerns with this implementation, because of firing a query per iteration. That's the main reason why in initial implementation there's a single query that fetches all existing data into an array.

Do you have any thoughts on this?